core/src/basic_config.rs

```diff
@@ -0,0 +1,27 @@
+use serde::{Deserialize, Deserializer};
+use shellexpand::full;
+use std::default::Default;
+use std::path::PathBuf;
+
+#[derive(Deserialize, Default, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct BasicConfigContent {
+    pub base: BaseConfig,
+}
+
+#[derive(Deserialize, Default, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct BaseConfig {
+    #[serde(deserialize_with = "deserialize_and_resolve_path")]
+    pub base_path: PathBuf,
+    pub _use_git_source_path: bool,
+}
+
+pub fn deserialize_and_resolve_path<'de, D>(deserializer: D) -> Result<PathBuf, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+    let Ok(expanded_path) = full(&s) else { todo!() };
```

Comment:
```suggestion
    let expanded_path) = full(&s)?;
```

the function signature already allows error -- i'd pass it through.

Reply:
<img width="1435" alt="image" src="https://github.com/user-attachments/assets/ace0c0f5-8042-4c15-8c2e-6154465e9692" />
Sadly doesn't seem possible :/ Or is there a way I don't know yet?